### PR TITLE
Reestructure the project as a python module.

### PR DESCRIPTION
**Pull request by agnjunio**
For now, assuming only compiledb.parser will be available. Also,
assuming that VERSION will contain the current version of the package,
since there is no definition of source of truth yet.

To test module installation:
* Pull branch
* (Recommended) Create a test project anywhere
* (Recommended) Create a new virtual env and activate it
* Run pip install -e <path to root> to install as dev mode
* Try to run the compiledb-gen-parser command as usual

Signed-off-by: Agnaldo Junior <agnaldo.junior01@gmail.com>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        'Intended Audience :: Developers',
+        'Topic :: Software Development :: Build Tools',
+        'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
```
Comment (nickdiego) [inline setup.py]: Currently we do not support python 3. I think it may be addressed in a separate issue soon.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    version = version_file.read().strip()
+
+setup(
+    name='compiledb',
+    version=version,
+    description='Tool for generating LLVM Compilation Database files for make-based build systems.',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    url='https://github.com/nickdiego/compiledb-generator',
+    author='Nick Diego',
```
Comment (nickdiego) [inline setup.py]: s/Diego/Yamane/ :)

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        'Programming Language :: Python :: 3.6',
+    ],
+    keywords='compilation-database clang c cpp makefile rtags completion',
+    packages=find_packages(exclude=['contrib', 'docs', 'tests']),
+    install_requires=[],
+    extras_require={
+        'dev': [],
+        'test': ['pytest'],
+    },
+    python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*',
```
Comment (nickdiego) [inline setup.py]: ditto.

File: setup.py (hunk 1 above)
Comment (agnjunio) [reply setup.py]: Removing then.

File: setup.py (hunk 2 above)
Comment (agnjunio) [reply setup.py]: s/Diego/Gay

**Pull request merged**